lib/ansible/plugins/action/win_reboot.py

```diff
@@ -174,18 +171,26 @@ def check_uptime():
 
             # reset the connection to clear the custom connection timeout
             try:
-                self._connection.set_options(direct={"connection_timeout": connection_timeout_orig})
+                self._connection.set_option("connecton_timeout", connection_timeout_orig)
                 self._connection._reset()
-            except (AnsibleError, AttributeError):
-                display.debug("Failed to reset connection_timeout back to default")
+            except AnsibleError as e:
+                display.debug("Failed to reset connection_timeout back to default: %s" % to_native(e))
 
             # finally run test command to ensure everything is working
             def run_test_command():
                 display.vvv("attempting post-reboot test command '%s'" % test_command)
-                (rc, stdout, stderr) = self._connection.exec_command(test_command)
-
-                if rc != 0:
-                    raise Exception('test command failed')
+                try:
+                    (rc, stdout, stderr) = self._connection.exec_command(test_command)
+                except Exception as e:
+                    # in case of a failure trying to execute the command
+                    # (another reboot occurred) we need to reset the connection
+                    # to make sure we are not re-using the same shell id
+                    self._connection._reset()
+                    raise e
```

Comment:
IIRC should be just `raise` to re-raise the existing error

Reply:
TIL, will try it out and change it.